pkg/apis/kubeone/types.go

```diff
@@ -111,6 +111,36 @@ type ClusterNetworkConfig struct {
 	ServiceSubnet     string `json:"serviceSubnet"`
 	ServiceDomainName string `json:"serviceDomainName"`
 	NodePortRange     string `json:"nodePortRange"`
+	CNI               *CNI   `json:"cni,omitempty"`
+}
+
+// CNIProvider type
+type CNIProvider string
+
+// List of CNI Providers
+const (
+	// CNIProviderCanal is a Canal CNI plugin (Flannel + Calico).
+	// Features:
+	// * Support Network Policies
+	// * Does not support traffic encryption
+	// More info: https://docs.projectcalico.org/v3.7/getting-started/kubernetes/installation/flannel
+	CNIProviderCanal CNIProvider = "canal"
+
+	// CNIProviderWeaveNet is a WeaveNet CNI plugin.
+	// Features:
+	// * Support Network Policies
+	// * Support optional traffic encryption
+	// * Strong secret will be autogenerated
+	// More info: https://www.weave.works/docs/net/latest/kubernetes/kube-addon/
```

Comment:
Is this the right link for `weave-net`? Would this one work better https://www.weave.works/oss/net/?

Reply:
yes, the link is correct.